source/backend/routes/users.js

```diff
@@ -59,31 +60,41 @@ router.get('/:id', async (req, res) => {
   }
 });
 
-/* DELETE /user/savedRecipes */
-router.delete('/:savedRecipeId', async (req, res) => {
+/* DELETE /users/savedRecipes/:savedRecipeId */
+router.delete('/savedRecipes/:savedRecipeId', verifyUserToken, async (req, res) => {
   try {
+    console.log('START!');
     const { userId } = req.userInfo;
-    const { recipeId } = req.recipeInfo;
-    if (parseInt(req.params.id, 10) !== userId) {
-      return res.status(401).json({ message: 'Forbidden, acceess denied' });
-    }
+    const { savedRecipeId } = req.params;
 
-    const row = await savedRecipesModel.getByUserIdAndRecipeId(userId, recipeId);
-    if (row.length <= 0) {
-      return res.status(404).json({ message: 'User information not found' });
-    }
-
-    await savedRecipesModel.removeById(row[0].savedRecipeId); 
-    return res.status(200).json("Removed successfully");
+    await savedRecipesModel.removeSavedRecipe(userId, savedRecipeId);
+    
+    return res.status(200).json({msg: 'Removed successfully'});
     
   } catch (err) {
     console.error(err);
     return res.status(503).json({
-      message: 'Failed to get user information due to'
+      msg: 'Failed to remove saved recipe from user\'s list due to'
         + 'internal server error',
       err,
     });
   }
 });
 
+/* POST /users/ */
+router.post('/savedRecipes/:recipeId', verifyUserToken, async (req, res) => {
+  try {
+    const { userId } = req.userInfo;
+    const { recipeId } = req.params;
+
```

Comment:
Similar to complete recipes, we want to make sure a user cannot store another saved recipe when that record already exists.

We need to do a check:
1. query savedRecipeModel with UserId and recipeId
2. if length is more than 0, return 401 forbidden
3. otherwise, proceed with adding.

Reply:
Working on this now

Reply:
Fixed